mistral/deployment/src/main/java/io/quarkiverse/langchain4j/mistralai/deployment/MistralAiProcessor.java

```diff
@@ -0,0 +1,102 @@
+    private static final String PROVIDER = "mistralai";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    public void providerCandidates(BuildProducer<ChatModelProviderCandidateBuildItem> chatProducer,
+            BuildProducer<EmbeddingModelProviderCandidateBuildItem> embeddingProducer,
+            LangChain4jMistralAiBuildConfig config) {
+        if (config.chatModel().enabled().isEmpty() || config.chatModel().enabled().get()) {
+            chatProducer.produce(new ChatModelProviderCandidateBuildItem(PROVIDER));
+        }
+        if (config.embeddingModel().enabled().isEmpty() || config.embeddingModel().enabled().get()) {
+            embeddingProducer.produce(new EmbeddingModelProviderCandidateBuildItem(PROVIDER));
+        }
+    }
+
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void generateBeans(MistralAiRecorder recorder,
+            List<SelectedChatModelProviderBuildItem> selectedChatItem,
+            List<SelectedEmbeddingModelCandidateBuildItem> selectedEmbedding,
+            LangChain4jMistralAiConfig config,
+            BuildProducer<SyntheticBeanBuildItem> beanProducer) {
+
+        for (var selected : selectedChatItem) {
+            if (PROVIDER.equals(selected.getProvider())) {
+                String modelName = selected.getModelName();
+                var builder = SyntheticBeanBuildItem
+                        .configure(CHAT_MODEL)
+                        .setRuntimeInit()
+                        .defaultBean()
+                        .scope(ApplicationScoped.class)
+                        .supplier(recorder.chatModel(config, modelName));
+                addQualifierIfNecessary(builder, modelName);
+                beanProducer.produce(builder.done());
+
+                var streamingBuilder = SyntheticBeanBuildItem
+                        .configure(STREAMING_CHAT_MODEL)
+                        .setRuntimeInit()
+                        .defaultBean()
+                        .scope(ApplicationScoped.class)
+                        .supplier(recorder.streamingChatModel(config, modelName));
+                addQualifierIfNecessary(streamingBuilder, modelName);
+                beanProducer.produce(streamingBuilder.done());
+            }
+        }
+
+        for (var selected : selectedEmbedding) {
+            if (PROVIDER.equals(selected.getProvider())) {
+                String modelName = selected.getModelName();
+                var builder = SyntheticBeanBuildItem
+                        .configure(EMBEDDING_MODEL)
+                        .setRuntimeInit()
+                        .defaultBean()
+                        .scope(ApplicationScoped.class)
+                        .supplier(recorder.embeddingModel(config, modelName));
+                addQualifierIfNecessary(builder, modelName);
+                beanProducer.produce(builder.done());
```

Comment:
Need to inject
```
BuildProducer<EmbeddingModelBuildItem> embeddingModelBuildItemBuildProducer,
```
as parameter, and then add here:
```
embeddingModelBuildItemBuildProducer.produce(new EmbeddingModelBuildItem());
```

to make it visible to the dev ui